src/main/java/com/sap/cloudfoundry/client/facade/util/AuthorizationEndpointGetter.java

```diff
@@ -0,0 +1,41 @@
+package com.sap.cloudfoundry.client.facade.util;
+
+import java.util.Map;
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+public class AuthorizationEndpointGetter {
+
+    private final WebClient webClient;
+
+    public AuthorizationEndpointGetter(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    public String getAuthorizationEndpoint(String controllerUrl) {
+        String response = webClient.get()
+                                   .uri(controllerUrl + "/")
+                                   .retrieve()
+                                   .bodyToMono(String.class)
+                                   .block();
+        return getLoginHref(response);
+    }
+
+    public String getAuthorizationEndpoint() {
+        String response = webClient.get()
+                                   .uri("/")
+                                   .retrieve()
+                                   .bodyToMono(String.class)
+                                   .block();
+        return getLoginHref(response);
+    }
```

Comment:
Remove the duplicated code from the second method and call the first with an empty string for `controllerUrl`